grease/src/Grease/Macaw/Dwarf.hs

```diff
@@ -0,0 +1,305 @@
+{-# LANGUAGE ImplicitParams #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | Dwarf shape parsing and initialization
```

Comment:
Thanks for moving this into its own module, this is really nice!

It's probably worth including a bit more content here in the module-level Haddock explaining the purpose of this module. Even something as simple as "Bootstrap the refinement loop by constructing initial 'Shape's in a type-directed way from DWARF debug info" would be helpful.